[BUG]: Numpy 2.4.0 breaks 6 cuda.core tests

### Is this a duplicate?

- [x] I confirmed there appear to be no [duplicate issues](https://github.com/NVIDIA/cuda-python/issues) for this bug and that I agree to the [Code of Conduct](CODE_OF_CONDUCT.md)

### Type of Bug

Something else

### Component

cuda.core

### Describe the bug

With Numpy 2.4.0, the `cuda.core` test suite fails on the following tests:

```
FAILED cuda_core/tests/test_utils.py::TestViewCPU::test_args_viewable_as_strided_memory_cpu[in_arr4] - assert (1,) is None
FAILED cuda_core/tests/test_utils.py::TestViewCPU::test_strided_memory_view_cpu_init[in_arr0] - assert (1,) is None
FAILED cuda_core/tests/test_utils.py::TestViewCPU::test_strided_memory_view_cpu[in_arr0] - assert (1,) is None
FAILED cuda_core/tests/test_utils.py::TestViewCPU::test_args_viewable_as_strided_memory_cpu[in_arr0] - assert (1,) is None
FAILED cuda_core/tests/test_utils.py::TestViewCPU::test_strided_memory_view_cpu_init[in_arr4] - assert (1,) is None
FAILED cuda_core/tests/test_utils.py::TestViewCPU::test_strided_memory_view_cpu[in_arr4] - assert (1,) is None
```

### How to Reproduce

Run the cuda.core tests with `Numpy 2.4.0` installed.  (Numpy 2.3.5 works).

### Expected behavior

Tests pass.

My hunch is that Numpy previously set `strides` to `None` when the data was C contiguous, but now it is setting it always.  I don't see a Numpy CHANGELOG entry for this, but `strides` is now readonly, fwiw.

### Operating System

All Python versions 3.11 and greater in CI.  This tracks, since 3.10 is not supported by Numpy 2.4.0, so there we get an earlier version installed.

### nvidia-smi output

_No response_

--- a/cuda_core/tests/test_utils.py
+++ b/cuda_core/tests/test_utils.py
@@ -95,10 +95,7 @@ class TestViewCPU:
         assert view.shape == in_arr.shape
         assert view.size == in_arr.size
         strides_in_counts = convert_strides_to_counts(in_arr.strides, in_arr.dtype.itemsize)
-        if in_arr.flags.c_contiguous:
-            assert view.strides is None or view.strides == strides_in_counts
-        else:
-            assert view.strides == strides_in_counts
+        assert (in_arr.flags.c_contiguous and view.strides is None) or view.strides == strides_in_counts
         assert view.dtype == in_arr.dtype
         assert view.device_id == -1
         assert view.is_device_accessible is False

